lib/iris/util.py

```diff
@@ -2145,3 +2146,51 @@ def _strip_metadata_from_dims(cube, dims):
             reduced_cube.remove_cell_measure(cm)
 
     return reduced_cube
+
+
+def mask_cube_from_shapefile(shape, cube, minimum_weight=0.0, in_place=False):
+    """Takes a shapefile and masks all points not touching it in a cube
+
+    Finds the overlap between the `shape` and the `cube` in 2D xy space and
+    masks out any cells with less % overlap with shape than set.
+    Default behaviour is to count any overlap between shape and cell as valid
+
+    Parameters
+    -----------
+
+    shape : Shapely.Geometry object
+        A single `shape` of the area to remain unmasked on the `cube`.
+        If it a line object of some kind then minimum_weight will be ignored,
+        because you cannot compare the area of a 1D line and 2D Cell
+    cube : :class:`~iris.cube.Cube` object
+        The `Cube` object to masked. Must be singular, rather than a `CubeList`
+    minimum_weight : float , default=0.0
+        A number between 0-1 describing what % of a cube cell area must
+        the shape overlap to include it.
+    in_place : bool, default=False
+        Whether to mask the `cube` in-place or return a newly masked `cube`.
+        Defaults to False.
+
+    Returns
+    --------
+
+    iris.Cube
+        A masked version of the input cube, if in_place is False
```

Comment:
I've done some experimentation with text formatting and sphinx domains, here is my suggestion:

```suggestion
    shape : Shapely.Geometry object
        A single `shape` of the area to remain unmasked on the `cube`.
        If it is a line object of some kind then minimum_weight will be 
        ignored, because you cannot compare the area of a 1D line and 2D Cell.
    cube : :class:`~iris.cube.Cube` object
        The :class:`~iris.cube.Cube` object to masked. Must be singular,
        rather than a :class:`~iris.cube.CubeList`
    minimum_weight : float , default=0.0
        A number between 0-1 describing what % of a `cube` cell area must
        the shape overlap to include it.
    in_place : bool, default=False
        Whether to mask the `cube` in-place or return a newly masked
        :class:`~iris.cube.Cube`.

    Returns
    --------

    :class:`iris.cube.Cube`
        A masked version of the input `cube`, if `in_place` is ``False``
```